src/foam/core/stdlib.js

```diff
@@ -361,8 +361,8 @@ foam.LIB({
     function clone(o) { return o; },
     function equals(a, b) { return a === b; },
     function compare(a, b) {
-      return ( b === null || b === undefined ) ? 1 :
-          a < b ? -1 : a > b ? 1 : 0;
+      return ( b === null || b === undefined || foam.Object.isInstance(b) ||
```

Comment:
Does this file `'use strict';`? If not, `undefined` could have been reassigned to anything. It may be sufficient to `'use strict';` this function.